### ISSUE TITLE
update param values after simulator is built when `scenario.name = "change-contacts"`

### ISSUE THREAD

**Issue by papsti** [phac-nml-phrsd/EPACmodel#42]
the `contact_changepoints` matrix gets inserted into the simulator when adding the `change-contacts` scenario

currently on the `update-params` branch, there is code to update the contact matrix and transmission vector if, _e.g._, we want to pass a different values$setting.weight to the simulator after it's already built. this doesn't update the baseline contact values to be used in the `change-contacts` scenario (before the intervention). to enable:

- [x] add to the `pf` update in `run_after_simulator.R` to be able to update the `contact_changepoints` matrix after the simulator is initialized
- [x] update `EPACmodel::simulate()` to be able to pass values for the `contact_changepoints` matrix

(this is the same as the development done previously on this branch to ensure that the values used for `state`, `flow`, `transmission.`, and `contact.` can all be updated after the simulator is built.)

once this is done and documentation of the new methods is complete on `update-params`, it can be merged into `main`.
